DynatraceAppMon/Context.cs

```diff
@@ -250,6 +256,86 @@ internal bool goToMethod(Lookup lookup)
             }
         }
 
+        private bool HasClassFiles(ProjectItem projectItem)
+        {
+            return (GetClassFiles(projectItem, true).Any());
+        }
+
+        private List<String> GetClassFiles(ProjectItem projectItem, bool stopOnFirstFound)
+        {
+            List<String> classFiles = new List<string>();
+            //Indexes of FileNames are from 1 to FileCount for the project item.
+            for (short i = 1; i <= projectItem.FileCount; i++)
+            {
+                string fileName = projectItem.FileNames[i];
+                string fileExtension = Path.GetExtension(fileName);
+                if (fileExtension != null && (fileExtension.EndsWith(".cs")))
+                {
+                    classFiles.Add(fileName);
+                    if (stopOnFirstFound) break;
+                }
+            }
+            return classFiles;
+        }
+
+        private List<String> GetClassFiles(ProjectItem projectItem)
+        {
+            return GetClassFiles(projectItem, false);
+        }
+
+        private CodeType FindCodeType(ProjectItems projectItems, string fullName)
```

Comment:
It's also 'recursive' search, in next method you use 'recursive' in method name and here not. Better to be consistent.